core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/datalake/DataLakeValidator.java

```diff
@@ -17,13 +17,17 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(DataLakeValidator.class);
 
     @Inject
-    private StackService stackService;
+    private SdxClientService sdxClientService;
 
     public void validate(Stack stack, ValidationResult.ValidationResultBuilder validationBuilder) {
         if (stack.getEnvironmentCrn() != null && StackType.DATALAKE.equals(stack.getType())) {
-            LOGGER.info("Get datalake count in environment {}", stack.getEnvironmentCrn());
-            Long datalakesInEnv = stackService.countByEnvironmentCrnAndStackType(stack.getEnvironmentCrn(), StackType.DATALAKE);
-            if (datalakesInEnv > 1L) {
+            LOGGER.info("Get data lake count in environment {}", stack.getEnvironmentCrn());
+            Long datalakesInEnv = sdxClientService.getByEnvironmentCrn(stack.getEnvironmentCrn())
+                    .stream()
+                    .filter(sdxClusterResponse -> !sdxClusterResponse.isDetached())
+                    .filter(sdxClusterResponse -> !sdxClusterResponse.getCrn().equals(stack.getResourceCrn()))
+                    .count();
+            if (datalakesInEnv >= 1L) {
```

Comment:
why have you changed this? I think cloudbreak still contains information about datalake, and sdxClientService requires additional API call which makes validation longer

Reply:
In case of re-sizing of the data lake, we are detaching the actual sdx and creating a new sdx with a different shape.

As far as CB is concerned, there is a data lake stack for the given environmentCrn. It does not anything about the sdx being dectached.

Reply:
I see your point, but still I would think about some solution which can make us avoid API calls. In this case we would call datalake service in case of every datalake creation which can slow down all datalake creation.

Reply:
side note: you have changed the if condition from `if (datalakesInEnv > 1L)` to `if (datalakesInEnv >= 1L)` which leads to validation error in every case I think

Reply:
back to the `DataLakeValidator` topixc: it is used in `CloudConfigValidationActions`, I am not sure for what purpose, but I think, even if we call datalake service and validating if there is only 1 live datalake there, in CB it can cause issues if there is 2 non terminated datalake stacks for an environment, so since @doktoric created mostly the `CloudConfigValidationActions` class, I would ask him, if what should be done here to not break anything during cloud config validation

Reply:
in general I think it can cause issues if we have 2 non-terminated datalake stacks for an environment at the same time, like here:
https://github.com/hortonworks/cloudbreak/blob/6439d4fe3ef62d7d61cb5f3f813bf466902ec9b8/core/src/main/java/com/sequenceiq/cloudbreak/repository/StackViewRepository.java#L28

Reply:
I aggree with @horadla23, you should check these kind of calls, it can cause troubles.

Reply:
In this case, the stack for the data lake which is detached is not in the terminated state but is in the stopped state.
Trying to understand what kind of issue might be possible?
let me reach out to Richard on this

Reply:
for example, this call will fail when an environment has a detached and a newer datalake: https://github.com/hortonworks/cloudbreak/blob/6439d4fe3ef62d7d61cb5f3f813bf466902ec9b8/core/src/main/java/com/sequenceiq/cloudbreak/repository/StackViewRepository.java#L28 (repository call will return more than one item)
this is just an example, but it was a general rule in CB that an environment has only 1 non-terminated datalake, but with this change in a small time interval there can be 2 non-terminated datalake for an environment.

Reply:
yes. There will be 2 non-terminated data lakes for short period.

Reply:
Let me summarise the problem here:
The current DH creation logic works on a way that some of the DL config inherited into the DH cluster. Currently we provision a DH into and environment and automatically assign it the to DL cluster which is running in the Environment. IF we have n+1 DL in the Environment we can not handle this scenario because we don't know which DL should we use.
Question regarding this scenario:
- Can we provision a DH cluster under DL resize process?
- Can it cause any issue on experience side if we have multiple DL in the environment ? 
Also one more thing, I think the UI side also can handle multiple DL in an environment.

Reply:
You are right, DH uses the data lake stack associated with the environment CRN to fetch the remote data context of DL.  If DH fetches remote data context of DL while DL re-sizing is in progress it will cause issues. DL re-size operation is performed after the data hub is stopped. One of the first steps in the re-size flow chain is to stop DL and DH clusters.

Service discovery internally depends on getByEnvCrn API expose by SdxEndpoint to the SDx information. We are filtering out data lakes that are detached before returning the list. we are good on that front as well.

Reply:
> One of the first steps in the re-size flow chain is to stop DL and DH clusters.

is this ensured somehow? and what about experiences? we need to make sure somehow that everything is stopped and we also need to make sure that it is not possible to start a new datahub during resize

Reply:
nvm,sdx stop will stop all datahubs